miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleRestController.java

```diff
@@ -314,7 +287,7 @@ public SampleDto updateSample(@PathVariable("id") Long id, @RequestBody SampleDt
     }
     List<Map<String, Set<SampleDto>>> identitiesBySearchTerm = new ArrayList<>();
     for (JsonNode term : searchTerms) {
-      Set<SampleDto> uniqueIdentities = getSamplesForIdentityString(term.asText(), project, false);
+      Set<SampleDto> uniqueIdentities = getSamplesForIdentityString(term.asText(), project, exactMatch);
       if (uniqueIdentities.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)